File: src/main/java/allane/model/Contract.java
package allane.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "contract")
public class Contract {
    @Id
    @GeneratedValue(strategy= GenerationType.SEQUENCE)
    @Column(name = "contract_id")
    private long contractId;
    private int contractNumber;
    private double monthlyRate;

    @ManyToOne
    private Customer customer;

    @ManyToOne
    private Vehicle vehicle;
}

File: src/web/src/app/model/contract.ts
import { Customer } from './customer';
import { Vehicle } from './vehicle';

export class Contract {
  contractId?: number;
  contractNumber?: number;
  monthlyRate?: number;
  customer?: Customer;
  vehicle?: Vehicle;

  constructor(
      contractId?: number,
      contractNumber?: number,
      monthlyRate?: number,
      customer?: Customer,
      vehicle?: Vehicle
    ) {
        this.contractId = contractId;
        this.contractNumber = contractNumber;
        this.monthlyRate = monthlyRate;
        this.customer = customer;
        this.vehicle = vehicle;
      }

}
